src/System.Globalization.Extensions/tests/Normalization/NormalizationAll.cs

```diff
@@ -14,192 +12,127 @@ public class NormalizationAll
+                
+                // Form KC
+                VerifyConformanceInvariant(NormalizationForm.FormKC, s_normalizationData[i][0], s_normalizationData[i][1], s_normalizationData[i][2], s_normalizationData[i][3], s_normalizationData[i][4]);
+                
+                // Form KD
+                VerifyConformanceInvariant(NormalizationForm.FormKD, s_normalizationData[i][0], s_normalizationData[i][1], s_normalizationData[i][2], s_normalizationData[i][3], s_normalizationData[i][4]);
+            }
+        }
 
-                    // Form KC
-                    failed |= VerifyConformanceInvariant(NormalizationForm.FormKC, s_normalizationData[i][0], s_normalizationData[i][1], s_normalizationData[i][2], s_normalizationData[i][3], s_normalizationData[i][4]);
+        /// <summary>
+        /// Verifies the first normalization conformance invariant for the
+        /// specified normalization form, where the rule as defined for all
+        /// Unicode normalization forms is as follows:
+        ///  1. The following invariants must be true for all conformant 
+        ///    implementations:
+        ///    NFC
+        ///      c2 ==  NFC(c1) ==  NFC(c2) ==  NFC(c3)
+        ///      c4 ==  NFC(c4) ==  NFC(c5)
+        ///
+        ///    NFD
+        ///      c3 ==  NFD(c1) ==  NFD(c2) ==  NFD(c3)
+        ///      c5 ==  NFD(c4) ==  NFD(c5)
+        ///
+        ///    NFKC
+        ///      c4 == NFKC(c1) == NFKC(c2) == NFKC(c3) == NFKC(c4) == NFKC(c5)
+        ///
+        ///    NFKD
+        ///      c5 == NFKD(c1) == NFKD(c2) == NFKD(c3) == NFKD(c4) == NFKD(c5)
+        /// </summary>
+        private static void VerifyConformanceInvariant(NormalizationForm normForm, string c1, string c2, string c3, string c4, string c5)
+        {
+            string normalized1 = c1.Normalize(normForm);
+            bool isNormalized1 = c1.IsNormalized(normForm);
 
-                    form = NormalizationForm.FormKD;
-                    // Form KD
-                    failed |= VerifyConformanceInvariant(NormalizationForm.FormKD, s_normalizationData[i][0], s_normalizationData[i][1], s_normalizationData[i][2], s_normalizationData[i][3], s_normalizationData[i][4]);
-                }
-                catch
-                {
-                    Console.WriteLine("Wrong index: {0} length = {1} form = {2}", i, s_normalizationData[i].Length, form);
-                    for (int j = 0; j < s_normalizationData[i].Length; j++ )
-                    {
-                        Console.WriteLine("'{0}'", DumpStringAsCodepoints(s_normalizationData[i][j]));
-                    }
-                    throw;
-                }
-            }
+            string normalized2 = c2.Normalize(normForm);
+            bool isNormalized2 = c2.IsNormalized(normForm);
```

Comment:
Some of these, like isNormalized2, appear to only be used in one of the cases... in which case it can be moved to that case rather than being created here.